### ISSUE TITLE
CSL styles not including annotations

### ISSUE THREAD

**Issue by K3das** [typst/hayagriva#91]
Trying the following entry with [`chicago-annotated-bibliography.csl`](https://github.com/citation-style-language/styles/blob/42f54b22d8058c957404945b7dca5cc59c1fc68d/chicago-annotated-bibliography.csl) results in no annotation being rendered, whereas the CSL editor renders this annotation.
```bibtex
@article{kameny_ladder,
	title = {Does Research Into Homosexuality Matter?},
	volume = {9},
	url = {https://digitalassets.lib.berkeley.edu/sfbagals/The_Ladder/1965_Ladder_Vol09_No08_May.pdf},
	pages = {14--21},
	number = {8},
	journaltitle = {The Ladder},
	author = {Kameny, Franklin},
	date = {1965-05},
	note = {Annotation test string},
}
```

[Here's an example project for reference](https://typst.app/project/rzIG3hP_8ZdrpptS886lg0)

The Citation Styles Visual Editor renders an annotation like so:
![image](https://github.com/typst/hayagriva/assets/20052500/60d7dadb-4874-4977-b9f4-27a01d9f9067)

Yet Typst has no annotation:
![image](https://github.com/typst/hayagriva/assets/20052500/06a58c91-2ca3-40f8-92da-ad7321e95864)

**Comment by DerDrodt** [comment]
The problem is that the translation from BibLaTeX to hayagriva throws out the note. When converting a `biblatex::Entry` to a `hayagriva::Entry`, it does the following:

```rs
if let Some(note) = map_res(entry.how_published())?.map(Into::into) {
    if let Some(parent) = book(&mut item, parent) {
        parent.set_note(note);
    } else {
        item.set_note(note);
    }
}
```

At no other point is `note` set.

The question therefore is, how to handle "how published": Give it its own hayagriva field or append it to the note?

**Comment by EpicEricEE** [comment]
I believe this should be fixed with #220. It now renders as

![image](https://github.com/user-attachments/assets/478b2067-f12b-44e2-8ec5-39ccb89b4b5a)
